convex/game/agents.ts

```diff
@@ -385,78 +351,53 @@ export const agentFindConversation = internalMutation({
-      .filter((q) => q.neq(q.field('_id'), args.playerId))
-      .collect();
+    // Don't try again if we recently tried to find someone to invite.
+    const recentlyAttempted =
+      agent.lastInviteAttempt && now < agent.lastInviteAttempt + CONVERSATION_COOLDOWN;
 
-    const candidates = [];
+    const recentActivity = player.activity && now < player.activity.until + ACTIVITY_COOLDOWN;
 
-    for (const otherPlayer of otherPlayers) {
-      // Skip players that are currently in a conversation.
-      const member = await conversationMember(ctx.db, otherPlayer._id);
-      if (member) {
-        continue;
-      }
-      // Find the latest conversation we're both members of.
-      const lastMember = await ctx.db
-        .query('conversationMembers')
-        .withIndex('left', (q) =>
-          q
-            .eq('playerId', args.playerId)
-            .eq('status.kind', 'left')
-            .eq('status.with', otherPlayer._id),
-        )
-        .order('desc')
-        .first();
-
-      if (lastMember) {
-        if (lastMember.status.kind !== 'left') {
-          throw new Error(`Unexpected status: ${lastMember.status.kind}`);
-        }
-        if (now < lastMember.status.ended + PLAYER_CONVERSATION_COOLDOWN) {
-          continue;
-        }
-      }
-      const location = await ctx.db.get(otherPlayer.locationId);
-      if (!location) {
-        throw new Error(`Couldn't find location: ${otherPlayer.locationId}`);
+    // Decide whether to do an activity or
+    if (!player.pathfinding) {
+      if (recentActivity || justLeftConversation) {
+        // TODO: decide where to go
+        // await insertInput(ctx, args.worldId, 'finishDoSomething', {
+        //   uuid: args.uuid,
+        //   agentId: args.agentId,
+        //   destination:
+        // });
+      } else {
+        // TODO: have LLM choose the activity & emoji
+        const activity = ACTIVITIES[Math.floor(Math.random() * ACTIVITIES.length)];
+        await insertInput(ctx, args.worldId, 'finishDoSomething', {
+          uuid: args.uuid,
+          agentId: args.agentId,
+          activity: {
+            description: activity.description,
+            emoji: activity.emoji,
+            until: Date.now() + activity.duration,
+          },
+        });
       }
-      const position = { x: location.x, y: location.y };
-      candidates.push({ _id: otherPlayer._id, position });
     }
 
-    // Sort by distance and take the nearest candidate.
-    candidates.sort((a, b) => distance(a.position, position) - distance(b.position, position));
-    const candidate = candidates.length > 0 ? candidates[0]._id : null;
+    const invitee =
+      justLeftConversation || recentlyAttempted
+        ? undefined
+        : await findConversationCandidate(ctx.db, player, args.worldId, now);
 
-    await insertInput(ctx, args.worldId, 'finishFindConversation', {
+    await insertInput(ctx, args.worldId, 'finishDoSomething', {
```

Comment:
it might be nice to make the decision an explicit enum here --
```ts
type Decision = 
  | { kind: "activity", description: string, ... }
  | { kind: "moveTo", destination: Point }
  | { kind: "invite", invitee: Id<"players"> }
  | { kind: "nothing" }
```
that way the "do nothing" case is a bit more explicit too.

Reply:
Yup. I was considering whether it might want to pass back multiple things at once. E.g. choosing where to go when inviting a user (e.g. walk to their house). I'll start loose and tighten it if it starts to get unweildly, but I'd like to reduce the number of places someone needs to make an edit to add behavior.